pkg/yurtadm/util/kubernetes/kubernetes.go

```diff
@@ -396,17 +396,25 @@ func SetKubeadmJoinConfig(data joindata.YurtJoinData) error {
 		"criSocket":              nodeReg.CRISocket,
 		"name":                   nodeReg.Name,
 	}
+
+	v1, err := version.NewVersion(data.KubernetesVersion())
+	if err != nil {
+		return err
+	}
+
 	if nodeReg.CRISocket == constants.DefaultDockerCRISocket {
 		ctx["networkPlugin"] = "cni"
 	} else {
-		ctx["containerRuntime"] = "remote"
+		v127, err := version.NewVersion("v1.27.0")
+		if err != nil {
+			return err
+		}
+		if v1.LessThan(v127) {
+			ctx["containerRuntime"] = "remote"
+		}
```

Comment:
Since this flag only needed for versions before 1.24, and has been deprecated in 1.24 through 1.26. So I think we only need this flag if the k8s version is earlier than `1.24.0-alpha.0`, what do you think?

Reply:
This flag has been deprecated in 1.24 through 1.26, but it is removed in 1.27. So it is ok to compare with version 1.27.0.

Reply:
OK, I got it.

Reply:
i fixed to compare with 1.24.0-alpha.0